jlm/llvm/opt/RvsdgTreePrinter.cpp

```diff
@@ -115,6 +118,78 @@ RvsdgTreePrinter::AnnotateNumRvsdgNodes(
   annotateRegion(*rvsdg.root());
 }
 
+void
+RvsdgTreePrinter::AnnotateNumMemoryStateInputsOutputs(
+    const rvsdg::graph & rvsdg,
+    util::AnnotationMap & annotationMap)
+{
+  static std::string_view argumentLabel("NumMemoryStateTypeArguments");
+  static std::string_view resultLabel("NumMemoryStateTypeResults");
+  static std::string_view inputLabel("NumMemoryStateTypeInputs");
+  static std::string_view outputLabel("NumMemoryStateTypeOutputs");
+
+  std::function<void(const rvsdg::region &)> annotateRegion = [&](const rvsdg::region & region)
```

Comment:
Is there a reason the lambda is stored as a function pointer in `std::function`, and not an `auto` functor?

Reply:
Ah, I see now it is recursive. Nice trick, I didn't know you could do recursive lambdas like that

Reply:
You answered it yourself. :)